proofs/refine.ml

```diff
@@ -55,14 +22,11 @@ let generic_refine ~typecheck f gl =
   (* Create the refinement term *)
   Proofview.Unsafe.tclEVARS sigma >>= fun () ->
   f >>= fun (v, c, principal) ->
-  Proofview.tclEVARMAP >>= fun sigma' ->
+  Proofview.tclEVARMAP >>= fun sigma ->
   Proofview.wrap_exceptions begin fun () ->
   (* Redo the effects in sigma in the monad's env *)
-  let privates_csts = Evd.eval_side_effects sigma' in
+  let privates_csts = Evd.eval_side_effects sigma in
   let env = Safe_typing.push_private_constants env privates_csts.Evd.seff_private in
-  (* Check that the introduced evars are well-typed *)
-  let fold accu ev = typecheck_evar ev env accu in
-  let sigma = if typecheck then Evd.fold_future_goals fold sigma' else sigma' in
   (* Check that the refined term is typesafe *)
   let sigma = if typecheck then Typing.check env sigma c concl else sigma in
```

Comment:
conclusion: if we do deep typechecking here (ie Typing.check instead of retyping + Typing.check_actual_type) it also makes sense to check future goals